### DISCUSSION TITLE
如何修改视频播放页面下面的按钮地址

### DISCUSSION THREAD

**Discussion by qwe771675022** [AlistGo/alist#1045] [Q&A] [answered]
请问，如何修改视频页面当中 ，视频播放器下面的几个按钮的地址
例如将 vlc按钮的地址：vlc://http://www.xxx.cn/xxx.flv 改成  http://www.xxx.cn/xxx.flv

**Comment by anwen-anyi** [comment, marked as answer]
1. css自己改（我没记错应该可以）
 2. 改源码
    - 源码地址
    - https://github.com/alist-org/alist-web/blob/524e80879a61a97ffbf6025a47460c565705eba2/src/pages/list/preview/video.tsx#L209

**Reply by qwe771675022** [reply]
我当前使用的是编译好的exe文件隶属于alist项目，请问如何将alist-web项目的内容共同编译在alist里面
我直接搭建php环境运行alist-web是没有任何页面的，请教—下如何使用

**Reply by anwen-anyi** [reply]
> 我当前使用的是编译好的exe文件隶属于alist项目，请问如何将alist-web项目的内容共同编译在alist里面 我直接搭建php换进运行alist-web是没有任何页面的，请教一下如何使用

第一个问题 - 如何将alist-web项目内容编译在alist里面这个文档写到了自己看吧
第二个问题 - 没明白是什么意思...alist前端是react写的php能跑么QAQ（没试过）,就算能跑没有后端也不行吧..

**Reply by qwe771675022** [reply]
感谢回答我的问题，alist-web已经编译成功了，
但是编译后端这个我没太看明白

clone https://github.com/Xhofe/alist 这个项目,将上一步dist目录下的文件全部拷贝至项目下的public目录，然后：

这个步骤我已经将编译好的文件拷贝进去了
从这里开始
appName="alist"
builtAt="$(date +'%F %T %z')"
goVersion=$(go version | sed 's/go version //')
gitAuthor=$(git show -s --format='format:%aN <%ae>' HEAD)
gitCommit=$(git log --pretty=format:"%h" -1)
gitTag=$(git describe --long --tags --dirty --always)
ldflags="\
-w -s \
-X 'github.com/Xhofe/alist/conf.BuiltAt=$builtAt' \
-X 'github.com/Xhofe/alist/conf.GoVersion=$goVersion' \
-X 'github.com/Xhofe/alist/conf.GitAuthor=$gitAuthor' \
-X 'github.com/Xhofe/alist/conf.GitCommit=$gitCommit' \
-X 'github.com/Xhofe/alist/conf.GitTag=$gitTag' \
"

到这里我没太看明白什么意思，在哪里执行
我直接执行了这条
go build -ldflags="$ldflags" alist.go
但是报错了
 go build -ldflags="$ldflags" alist.go
# command-line-arguments
C:\Program Files\Go\pkg\tool\windows_amd64\link.exe: running gcc failed: exit status 1
c:/mingw/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/4.8.2/../../../../x86_64-w64-mingw32/bin/ld.exe: unrecognized option '--high-entropy-va'
c:/mingw/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/4.8.2/../../../../x86_64-w64-mingw32/bin/ld.exe: use the --help option for usage information
collect2.exe: error: ld returned 1 exit status
可能是我没有理解上面那个是在哪里输入的，请教一下上面那些内容编译的时候在哪里输入的

**Reply by anwen-anyi** [reply]
> 感谢回答我的问题，alist-web已经编译成功了， 但是编译后端这个我没太看明白
> 
> clone https://github.com/Xhofe/alist 这个项目,将上一步dist目录下的文件全部拷贝至项目下的public目录，然后：
> 
> 这个步骤我已经将编译好的文件拷贝进去了 从这里开始 appName="alist" builtAt="$(date +'%F %T %z')" goVersion=$(go version | sed 's/go version //') gitAuthor=$(git show -s --format='format:%aN <%ae>' HEAD) gitCommit=$(git log --pretty=format:"%h" -1) gitTag=$(git describe --long --tags --dirty --always) ldflags=" -w -s -X 'github.com/Xhofe/alist/conf.BuiltAt=$builtAt' -X 'github.com/Xhofe/alist/conf.GoVersion=$goVersion' -X 'github.com/Xhofe/alist/conf.GitAuthor=$gitAuthor' -X 'github.com/Xhofe/alist/conf.GitCommit=$gitCommit' -X 'github.com/Xhofe/alist/conf.GitTag=$gitTag' "
> 
> 到这里我没太看明白什么意思，在哪里执行 我直接执行了这条 go build -ldflags="$ldflags" alist.go 但是报错了 go build -ldflags="$ldflags" alist.go
> 
> # command-line-arguments
> C:\Program Files\Go\pkg\tool\windows_amd64\link.exe: running gcc failed: exit status 1 c:/mingw/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/4.8.2/../../../../x86_64-w64-mingw32/bin/ld.exe: unrecognized option '--high-entropy-va' c:/mingw/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/4.8.2/../../../../x86_64-w64-mingw32/bin/ld.exe: use the --help option for usage information collect2.exe: error: ld returned 1 exit status 可能是我没有理解上面那个是在哪里输入的，请教一下上面那些内容编译的时候在哪里输入的

编译后端我看你是 *Windows* 版的 编译命令要改下,   改成   **go build -ldflags="-w -s" alist.go**
或者看我之前纪录的视频 ：https://b23.tv/2uuBNgH
